modules/sitemaps/sitemaps.php

```diff
@@ -1,792 +1,420 @@
- */
-function jetpack_sitemap_array_to_simplexml( $data, &$tree ) {
-	$doc_namespaces = $tree->getDocNamespaces();
-
-	foreach ( $data as $key => $value ) {
-		// Allow namespaced keys by use of colon in $key, namespaces must be part of the document
-		$namespace = null;
-		if ( false !== strpos( $key, ':' ) && 'image' != $key ) {
-			list( $namespace_prefix, $key ) = explode( ':', $key );
-			if ( isset( $doc_namespaces[ $namespace_prefix ] ) ) {
-				$namespace = $doc_namespaces[ $namespace_prefix ];
-			}
-		}
+	/**
+	 * @see Jetpack_Sitemap_Librarian
+	 * @since 4.5.0
+	 * @var Jetpack_Sitemap_Librarian $librarian Librarian object for storing and retrieving sitemap data.
+	 */
+	private $librarian;
 
-		if ( 'image' != $key ) {
-			if ( is_array( $value ) ) {
-				$child = $tree->addChild( $key, null, $namespace );
-				jetpack_sitemap_array_to_simplexml( $value, $child );
-			} else {
-				$tree->addChild( $key, esc_html( $value ), $namespace );
-			}
-		} elseif ( is_array( $value ) ) {
-			foreach ( $value as $image ) {
-				$child = $tree->addChild( $key, null, $namespace );
-				jetpack_sitemap_array_to_simplexml( $image, $child );
-			}
-		}
-	}
+	/**
+	 * @see Jetpack_Sitemap_Logger
+	 * @since 4.5.0
+	 * @var Jetpack_Sitemap_Logger $logger Logger object for reporting debug messages.
+	 */
+	private $logger;
 
-	return $tree;
-}
+	/**
+	 * @see Jetpack_Sitemap_Finder
+	 * @since 4.5.0
+	 * @var Jetpack_Sitemap_Finder $finder Finder object for dealing with sitemap URIs.
+	 */
+	private $finder;
 
-/**
- * Define an array of attribute value pairs for use inside the root element of an XML document.
- * Intended for mapping namespace and namespace URI values.
- * Passes array through jetpack_sitemap_ns for other functions to add their own namespaces.
- *
- * @module sitemaps
- *
- * @return array array of attribute value pairs passed through the jetpack_sitemap_ns filter
- */
-function jetpack_sitemap_namespaces() {
 	/**
-	 * Filter the attribute value pairs used for namespace and namespace URI mappings.
-	 *
-	 * @module sitemaps
-	 *
-	 * @since 3.9.0
+	 * Construct a new Jetpack_Sitemap_Manager.
 	 *
-	 * @param array $namespaces Associative array with namespaces and namespace URIs.
+	 * @since 4.5.0
 	 */
-	return apply_filters( 'jetpack_sitemap_ns', array(
-		'xmlns:xsi'          => 'http://www.w3.org/2001/XMLSchema-instance',
-		'xsi:schemaLocation' => 'http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd',
-		'xmlns'              => 'http://www.sitemaps.org/schemas/sitemap/0.9',
-		// Mobile namespace from http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
-		'xmlns:mobile'       => 'http://www.google.com/schemas/sitemap-mobile/1.0',
-		'xmlns:image'        => 'http://www.google.com/schemas/sitemap-image/1.1',
-	) );
-}
+	public function __construct() {
+		$this->librarian = new Jetpack_Sitemap_Librarian();
+		$this->logger = new Jetpack_Sitemap_Logger();
```

Comment:
Maybe this can only be instantiated when `WP_DEBUG` is defined and set to true? It's better not to load this every time during production.